### PR TITLE
Mkt 6648

### PR DESCRIPTION
**Pull request by Contentstack-AnkitaD**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=zD0diL1)

### CODE REVIEW

File: src/fetch.js [hunk 1]
```diff
@@ -260,34 +262,52 @@ const getSyncData = async (
...
     const aggregatedSyncToken = syncToken.filter(item => item !== undefined);
     for (const token of aggregatedSyncToken) {
       const syncResponse = await fetchCsData(
         url,
         config,
-        (query = { sync_token: token })
-      );
-      aggregatedResponse.data = aggregatedResponse.data?.concat(
-        ...syncResponse.items
+        { sync_token: token }
```
Comment (abhishek305) [inline src/fetch.js:300]: why query was removed ?

**Review by rijil-tr — APPROVED** [review]

File: src/fetch.js [hunk 2]
```diff
@@ -260,34 +262,52 @@ const getSyncData = async (
...
       );
+      aggregatedResponse.data = aggregatedResponse.data.concat(syncResponse.items);
```
Comment (abhishek305) [inline src/fetch.js:302]: this is where we aggregate all the data make sure no entries are missing while quering via graphql.
I suspect removing spread operator would result in data loss.

**Review by abhishek305 — CHANGES_REQUESTED** [review]
Note : 

- Fixes snyk issues
- check the spread operator 
- test it with extensive stacks like small and big ones for data discrepancies

File: src/fetch.js [hunk 3]
```diff
@@ -226,16 +226,18 @@ const getPagedData = async (
   return aggregatedResponse;
 };
 
-const getSyncData = async (
+exports.getSyncData = async (
```
[the remark this answers is not in the document]
Comment (Contentstack-AnkitaD) [reply src/fetch.js:229]: No we arent pushing the unit test cases. as that is not complete and. doesnt fit into the scope of the ticket.

File: src/fetch.js (hunk 1 above)
Comment (Contentstack-AnkitaD) [reply src/fetch.js:300]: Query wasnt removed. it was just refactored.. let me restore it back to previous state.. It wont make any difference as such

**Comment by Contentstack-AnkitaD** [comment]
> Note :
> 
> * Fixes snyk issues
> * check the spread operator
> * test it with extensive stacks like small and big ones for data discrepancies

Found No Issues with Direct Upgrade in synk test.

**Pull request merged**